modules/percolator/src/test/java/org/elasticsearch/percolator/PercolatorFieldMapperTests.java

```diff
@@ -480,9 +489,14 @@ public void testUnsupportedQueries() {
     }
 
     private void assertQueryBuilder(BytesRef actual, QueryBuilder expected) throws IOException {
-        XContentParser sourceParser = createParser(PercolatorFieldMapper.QUERY_BUILDER_CONTENT_TYPE.xContent(),
-                new BytesArray(actual));
-        assertThat(parseInnerQueryBuilder(sourceParser), equalTo(expected));
+        try (InputStream in = new ByteArrayInputStream(actual.bytes, actual.offset, actual.length)) {
+            try (StreamInput input = new NamedWriteableAwareStreamInput(new InputStreamStreamInput(in), writableRegistry())) {
+                input.readVInt();
+                input.readVInt();
```

Comment:
I will add a comment. The reason two vints are read is because of the format for binary fields.